library/src/main/java/com/bumptech/glide/manager/PostMConnectivityMonitor.java

```diff
@@ -0,0 +1,161 @@
+    this.listener = listener;
+    callback =
+        new ConnectivityManager.NetworkCallback() {
+          @Override
+          public void onAvailable(@NonNull Network network) {
+            onConnectivityChanged(true);
+          }
+
+          @Override
+          public void onLost(@NonNull Network network) {
+            onConnectivityChanged(false);
+          }
+
+          @Override
+          public void onCapabilitiesChanged(
+              @NonNull Network network, @NonNull NetworkCapabilities networkCapabilities) {
+            boolean isConnected = checkNetworkConnected(networkCapabilities);
+            onConnectivityChanged(isConnected);
+          }
+        };
+  }
+
+  @SuppressWarnings("WeakerAccess")
+  @Synthetic
+  void onConnectivityChanged(boolean isConnected) {
+    boolean wasConnected = this.isConnected;
+    this.isConnected = isConnected;
+    if (wasConnected != isConnected) {
+      if (Log.isLoggable(TAG, Log.DEBUG)) {
+        Log.d(TAG, "connectivity changed, isConnected: " + isConnected);
+      }
+
+      listener.onConnectivityChanged(isConnected);
+    }
+  }
+
+  @SuppressLint("MissingPermission")
+  public void register() {
+    if (isRegistered) {
+      return;
+    }
+    ConnectivityManager connectivityManager =
+        Preconditions.checkNotNull(
+            (ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE));
+    // Initialize isConnected.
+    isConnected = isConnected(connectivityManager);
+
+    try {
+      if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
+        connectivityManager.registerDefaultNetworkCallback(callback);
+      } else {
+        NetworkRequest request = new NetworkRequest.Builder().build();
+        connectivityManager.registerNetworkCallback(request, callback);
+      }
+      isRegistered = true;
+    } catch (RuntimeException e) {
+      if (Log.isLoggable(TAG, Log.WARN)) {
+        Log.w(TAG, "Failed to register", e);
+      }
+    }
+  }
+
+  public void unregister() {
+    if (!isRegistered) {
+      return;
+    }
+    ConnectivityManager connectivityManager =
+        Preconditions.checkNotNull(
+            (ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE));
+    try {
+      connectivityManager.unregisterNetworkCallback(callback);
+    } catch (RuntimeException e) {
+      // callback was registered more than once
+      if (Log.isLoggable(TAG, Log.WARN)) {
+        Log.w(TAG, "already unregister", e);
+      }
+    }
+    isRegistered = false;
+  }
+
+  @SuppressWarnings("WeakerAccess")
+  @Synthetic
+  // Permissions are checked in the factory instead.
+  @SuppressLint("MissingPermission")
+  boolean isConnected(@NonNull ConnectivityManager connectivityManager) {
+    NetworkCapabilities networkCapabilities;
+    try {
+      networkCapabilities =
+          connectivityManager.getNetworkCapabilities(connectivityManager.getActiveNetwork());
+    } catch (RuntimeException e) {
```

Comment:
Why is it necessary to make such a broad catch of RuntimeException here? At a minimum, could you add a comment on why here? Even better would be if we could slim this down to the specific exception we're concerned about (and probably document that as well).

Reply:
getNetworkCapabilities may throw RemoteException that was wrapped by RuntimeException